File: src/main/java/com/freya02/botcommands/components/internal/HandleComponentResult.java
package com.freya02.botcommands.components.internal;

import com.freya02.botcommands.components.ComponentErrorReason;
import org.jetbrains.annotations.Nullable;

public class HandleComponentResult {
	private final ComponentErrorReason errorReason;
	private final boolean shouldDelete;

	public HandleComponentResult(ComponentErrorReason errorReason, boolean shouldDelete) {
		this.errorReason = errorReason;
		this.shouldDelete = shouldDelete;
	}

	/**
	 * Whether additional resources (such as the components handler maps) should get cleaned up
	 */
	public boolean shouldDelete() {
		return shouldDelete;
	}

	@Nullable
	public ComponentErrorReason getErrorReason() {
		return errorReason;
	}
}

File: src/main/kotlin/com/freya02/botcommands/internal/components/HandleComponentResult.kt
package com.freya02.botcommands.internal.components

import com.freya02.botcommands.api.components.ComponentErrorReason

internal class HandleComponentResult(
    val errorReason: ComponentErrorReason?,
    /**
     * Whether additional resources (such as the components handler maps) should get cleaned up
     */
    val shouldDelete: Boolean
)